# Reason: We only want to build man pages, not development docs
# Upstream: Patch was rejected - https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/689
diff --git a/meson.build b/meson.build
index 1cc11d0dfdc..b25b9d93554 100644
--- a/meson.build
+++ b/meson.build
@@ -977,10 +977,8 @@ if enable_qt
 endif
 
 if enable_docs
-  assert(enable_introspection, '-Ddocs=true requires -Dintrospection=true')
   assert(meson.version().version_compare('>= 0.49.0'), '-Ddocs requires meson >= 0.49')
   subdir('man')
-  subdir('docs')
   meson.add_dist_script(
     'tools/meson-dist-data.sh',
     '--build-root', build_root
diff --git a/src/libnmc-setting/meson.build b/src/libnmc-setting/meson.build
index 7fb460dc331..fadc90f01f8 100644
--- a/src/libnmc-setting/meson.build
+++ b/src/libnmc-setting/meson.build
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: LGPL-2.1-or-later
 
-if enable_docs
+if enable_docs and enable_introspection
   assert(enable_introspection, '-Ddocs=true requires -Dintrospection=true')
 
   merge_cmd = join_paths(meson.source_root(), 'tools', 'generate-docs-nm-settings-docs-merge.py')
